lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,275 @@
+    #
+    # Examples
+    #   Response if success: [
+    #     {"id":"4","name":"Test Section","description":"A Test Section"},
+    #     {"id":"5","name":"Awesome Section","description":"A totally awesome Section"}
+    #   ]
+    #   Response if no sections exist:
+    #     {"code": 200, "success":true, "message": "No sections available", "time": 0.004, "data": []}
+    #   Response if sections are not supported:
+    #     {"code": 501, "error": "Sections are not supported"}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_groups
+      parse get("/groups")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain groups from External IPAM."))
+    end
+
+    # Get a group from External IPAM. A group is analagous to a 'section' in phpIPAM, and
+    # is a logical grouping of subnets/ips.
+    #
+    # Params: group_name: The name of the External IPAM group
+    #
+    # Returns: An array of sections on success, or a hash with a "message" key
+    #          containing error on failure.
+    #
+    # Examples
+    #   Response if success:
+    #     {"id":"5","name":"Awesome Section","description":"A totally awesome Section"}
+    #   Response if no sections exist:
+    #     {"code": 200, "success":true, "message": "No sections available", "time": 0.004, "data": []}
+    #   Response if sections are not supported:
+    #     {"code": 501, "error": "Sections are not supported"}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_group(group_name)
+      raise "group_name must be provided" if group_name.blank?
+      parse get("/groups/#{URI.escape(group_name)}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain group '#{group_name}' from External IPAM."))
+    end
+
+    # Get a list of subnets for the given External IPAM section.
+    #
+    # Params:  1. section_name: The name of the External IPAM section
+    #
+    # Returns: An array of subnets on success, or a hash with a "message" key
+    #          containing error on failure.
+
+    # Examples:
+    #   Response if success: {"code": 200, "success": true, "data": [
+    #     {"id":"9","subnet":"100.10.10.0","mask":"24","sectionId":"5","description":"Test Subnet 1"},
+    #     {"id":"10","subnet":"100.20.20.0","mask":"24","sectionId":"5","description":"Test Subnet 2"}
+    #   ],"time":0.018}
+    #   Response if no subnets exist in section.
+    #     {"code": 200, "success": true, "data": [], "time": 0.007}
+    #   Response if section not found:
+    #     {"code": 404, "error": "Section not found in External IPAM."}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_subnets_by_group(group_name)
```

Comment:
Again, id and sectionId. If your plugin will keep returning these it's not a problem, just delete those from documentation and keep returning them - Foreman will ignore them, unless I am not mistaken and they are used somewhere in the code.